src/servers/trigger.ts

```diff
@@ -137,6 +142,39 @@ export async function onchainTransactionEventHandler(tx) {
   }
 }
 
+export async function onchainBlockEventhandler({ lnd, height }) {
+  // get incoming txs from the last 36 blocks ~6 hours
+  // we don't use default look back because daily cron job will take care of delayed transactions
+  const lookBack = 6 * 6 // hours * ~ blocks per hour
+  const onchainTxns = await getOnChainTransactions({ lnd, lookBack, incoming: true })
+  const hasTransactions = onchainTxns && onchainTxns.length
+
+  if (!hasTransactions) {
+    logger.info(`finish block ${height} handler without transactions`)
+    return
+  }
+
+  await runInParallel({
+    iterator: onchainTxns.values(),
+    logger,
+    workers: 5,
+    processor: async (tx, index) => {
+      logger.trace("updating onchain tx %o in worker %d", tx.id, index)
+
+      const user = await User.findOne({ "onchain.address": { $in: tx.output_addresses } })
```

Comment:
we need to validate performance of this query

Reply:
if we have an index that shouldn't be a problem (not sure we have one)